clippy_lints/src/matches/manual_filter.rs

```diff
@@ -0,0 +1,191 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::ty::is_type_diagnostic_item;
+use clippy_utils::{is_lang_ctor, path_to_local_id};
+use rustc_hir::intravisit::{walk_expr, Visitor};
+use rustc_hir::LangItem::OptionSome;
+use rustc_hir::{Arm, Block, BlockCheckMode, Expr, ExprKind, HirId, Pat, PatKind, UnsafeSource};
+use rustc_lint::LateContext;
+use rustc_span::{sym, SyntaxContext};
+
+use super::manual_map::{check_with, SomeExpr};
+use super::MANUAL_FILTER;
+
+#[derive(Default)]
+struct NeedsUnsafeBlock(pub bool);
+
+impl<'tcx> Visitor<'tcx> for NeedsUnsafeBlock {
+    fn visit_expr(&mut self, expr: &'tcx Expr<'tcx>) {
+        match expr.kind {
+            ExprKind::Block(
+                Block {
+                    rules: BlockCheckMode::UnsafeBlock(UnsafeSource::UserProvided),
+                    ..
+                },
+                _,
+            ) => {
+                self.0 = true;
+            },
+            _ => walk_expr(self, expr),
+        }
+    }
+}
```

Comment:
We have a check for unsafe block in the util, maybe we can re-use that for this? https://github.com/rust-lang/rust-clippy/blob/e43f5a17d4d9263616ae6fc28a7c5353e80d8a12/clippy_utils/src/visitors.rs#L446-L475

Reply:
perfect, thanks for the suggestion 👍